### DISCUSSION TITLE
Implementing the MEF NRP YANG modules

### DISCUSSION THREAD

**Discussion by jahveen** [MEF-GIT/MEF-LSO-Presto-SDK#23] [Team Posts]
Hello,

As stated in the SDK (R3), the SDK "Contains the MEF NRP YANG modules, including the ONF Transport API yang modules on which the MEF NRP API is based". For an implementation, there are numerous attributes that a server will not use - for example, both the TAPI and MEF NRP YANG modules provide attributes for bandwidth profiles, and a server implementation will unlikely implement both.

RFC 7950 (The YANG 1.1 Data Modeling Language) allows a server to document unsupported portions of a base module using the "Deviation" statement. Is this also what is recommended in the community as a solution to document unused attributes in the YANG modules?

Thanks.

Jahveen

**Comment by MichalLaczynskiAmartus** [comment]
Hi Jahveen,

The answer I've managed to get is as follows:
To my knowledge, this YANG “Deviation” statement was never discussed in MEF NRM/NRP and ONF TAPI.
Same in the IISOMI.
In fact, all these modeling groups are more focused on UML development. This of course does not prevent starting a discussion on the topic – as YANG is becoming central…

Hope that helps,
Michał

**Comment by rgaglian** [comment]
I have tested a number of TAPI vendors and deviations are frequent, period. Documenting them should be encouraged in order to make integration possible. Deviations should also be published at the yang-catalog so a northbound system could get them in a GET call.

**Comment by jahveen** [comment]
Thanks for your responses Michal and rgaglian. I have a couple of follow-up questions:

@MichalLaczynskiAmartus - for MEF NRM/NRP, on what forum do you recommend that I start a discussion topic?

@rgaglian - I searched the yang-catalog for examples of deviation modules, and there only seem to be a handful published by Cisco. Would I be correct in saying that most vendors have chosen not to publish theirs?

Thanks,
Jahveen

**Comment by MichalLaczynskiAmartus** [comment]
Well this is the best place I think.

**Comment by jahveen** [comment]
Thanks @MichalLaczynskiAmartus. Then I think we can continue the discussion in this thread.

**Comment by rgaglian** [comment]
There are a number of deviations published here:
https://github.com/YangModels/yang

On Thu, Sep 10, 2020 at 11:32 AM jahveen <notifications@github.com> wrote:

> Thanks for your responses Michal and rgaglian. I have a couple of
> follow-up questions:
>
> @MichalLaczynskiAmartus <https://github.com/MichalLaczynskiAmartus> - for
> MEF NRM/NRP, on what forum do you recommend that I start a discussion topic?
>
> @rgaglian <https://github.com/rgaglian> - I searched the yang-catalog for
> examples of deviation modules, and there only seem to be a handful
> published by Cisco. Would I be correct in saying that most vendors have
> chosen not to publish theirs?
>
> Thanks,
> Jahveen
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/MEF-GIT/teams/mef-lso-presto-sdk-community/discussions/6/comments/3>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAS7IM7ELQD44N6WDC2QHRLSFCMKRANCNFSM4QEYXVBQ>
> .
>


-- 


At least I did something
Don Draper - Mad Men

**Comment by karthik-sethuraman** [comment]
Hi

While this does not address the deviation question directly, I would like to point you to the recently published ONF TR-547 (TAPI Reference Implementation Agreement) available here https://www.opennetworking.org/wp-content/uploads/2020/08/TR-547-TAPI-v2.1.3-Reference-Implementation-Agreement-1.pdf

This could potentially be one item for consideration in the next revision of the document.

Regards
Karthik

**Comment by jahveen** [comment]
Hello Karthik,

I would certainly welcome offical guidance on this area. Based on the community responses, I will assume that deviation modules are the recommended method of documenting unused attributes.

Thanks,
Jahveen
